Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
    run(1)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
Get loader
Files already downloaded and verified
x shape is (40000, 32, 32, 3)
Traceback (most recent call last):
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 248, in <module>
    run(1)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 243, in run
    ggn_vp(loader,state,mesh_sharding,sharding, global_shape_x)
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 119, in ggn_vp
    ggn_vp += jax.lax.scan(body_fn, init_carry, x)[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/project_465001020/hackathon/lumi_hackathon/scripts/dataparallelism_multinode_ggn.py", line 112, in body_fn
    x_ = jax.make_array_from_single_device_arrays(global_shape_x, x_sharding, x_arrays)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda3/envs/jax/lib/python3.12/site-packages/jax/_src/array.py", line 810, in make_array_from_single_device_arrays
    raise ValueError(
ValueError: jax.make_array_from_single_device_arrays requires a list of concrete arrays as input. got types {<class 'jax._src.interpreters.partial_eval.DynamicJaxprTracer'>}
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
srun: error: nid006322: tasks 8-15: Exited with exit code 1
srun: Terminating StepId=8180955.0
srun: error: nid006321: tasks 0-7: Exited with exit code 1
